File: Array_Functions/Count_Occurence/count_occurence.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

template <typename T>
unordered_map<T, int> Count_Occurences(vector<T> arr)
{
    unordered_map<T,int> mpp;
    for (T i: arr)
        mpp[i]++;
    
    return mpp;
}

int main()
{
    vector<int> vc = {1,2,3,5,4,4,6,3,2,3,7};
    unordered_map<int, int> mpp = Count_Occurences(vc);
    for(auto it: mpp)
    {
        cout << it.first << " " << it.second << endl;
    }
    return 0;
}

File: Array_Functions/Count_Occurence/count_occurence.py
def Count_Occurences(arr):
    mpp = {}
    for i in arr:
        mpp[i] = mpp.get(i,0) + 1

    return mpp

if __name__ == "__main__":
    arr = [1,2,3,3,5,4,6,4,1,2]
    mpp = Count_Occurences(arr)
    for key, value in mpp.items():
        print(key, value)
